Source/FixedBackground/BoostedBHFixedBG.hpp

```diff
@@ -0,0 +1,244 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef BOOSTEDBHFIXEDBG_HPP_
+#define BOOSTEDBHFIXEDBG_HPP_
+
+#include "ADMFixedBGVars.hpp"
+#include "Cell.hpp"
+#include "Coordinates.hpp"
+#include "DimensionDefinitions.hpp"
+#include "Tensor.hpp"
+#include "TensorAlgebra.hpp"
+#include "UserVariables.hpp"
+#include "simd.hpp"
+
+//! Class which computes the initial conditions for a boosted BH
+// in isotropic Schwazschild coords
+// Undoosted elem. ds^2 = -Adt + B dx^2
+// Lorentz boost x = \gamma (x' - vt'), t = \gamma (t' - v*x')
+// Addisional shift to fix coordinates to the BH
+// x' = \tilde x + v \tilde t, t' = \tilde t
+// BH appears fixed, but we're in the rest frame of the SF!
+
+class BoostedBHFixedBG
+{
+  public:
+    //! Struct for the params of the  BH
+    struct params_t
+    {
+        double mass = 1.0;                      //!<< The mass of the BH
+        std::array<double, CH_SPACEDIM> center; //!< The center of the BH
+        double velocity = 0.0; //!< The boost velocity in the x direction
+    };
+
+    template <class data_t> using Vars = ADMFixedBGVars::Vars<data_t>;
+
+    const params_t m_params;
+    const double m_dx;
+
+    BoostedBHFixedBG(params_t a_params, double a_dx)
+        : m_params(a_params), m_dx(a_dx)
+    {
+        // check this boost param is sensible
+        if ((m_params.velocity > 1.0) || (m_params.velocity < -1.0))
+        {
+            MayDay::Error("The boost velocity parameter must be in the range "
+                          "-1.0 < velocity < 1.0");
+        }
+    }
+
+    /// This just calculates chi which helps with regridding, debug etc
+    /// it is only done once on setup as the BG is fixed
+    template <class data_t> void compute(Cell<data_t> current_cell) const
+    {
+        // get position and set vars
+
+        Vars<data_t> metric_vars;
+        compute_metric_background(metric_vars, current_cell);
+
+        // calculate and save chi
+        data_t chi = TensorAlgebra::compute_determinant_sym(metric_vars.gamma);
+        chi = pow(chi, -1.0 / 3.0);
+
+        current_cell.store_vars(chi, c_chi);
+    }
+
+    /// Schwarzschild boosted solution as above
+    template <class data_t, template <typename> class vars_t>
+    void compute_metric_background(vars_t<data_t> &vars,
+                                   const Cell<data_t> &current_cell) const
```

Comment:
Change function so it passes Coordinates object rather than Cell - this allows it to be used more easily independently of BoxLoops.